pkg/approver/aws/aws.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type awsCloud struct {
+	ec2 *ec2.EC2
+	asg *autoscaling.AutoScaling
+}
+
+func newAWSCloud(regionName string) (*awsCloud, error) {
+	creds := credentials.NewChainCredentials([]credentials.Provider{
+		&credentials.EnvProvider{},
+		&credentials.SharedCredentialsProvider{},
+		&ec2rolecreds.EC2RoleProvider{
+			Client: ec2metadata.New(session.New(&aws.Config{})),
+		},
+	})
+
+	if regionName == "" {
```

Comment:
If you're looking stuff up in the metadata service, wouldn't the credentials come from there as well? Kinda like Kubernetes in cluster client?

Reply:
EC2 and ASG clients need region name to query. therefore either explicitly specify it, or we can get it from metadata service assuming. might be helpful when running from outside the cluster.

Reply:
This should only be applied if the `EC2Role` credential provider was used then, right?

Reply:
not necessarily.
EC2 and ASG resources are region scoped. so the region name is used to scope the instance or asg you are looking for in the api request.

creds tell you if you have access to api you are requesting.